rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/ldap/LdapClient.java

```diff
@@ -403,4 +413,25 @@ public AccessIdRepresentationModel doMapFromContext(final DirContextOperations c
       return accessId;
     }
   }
+
+  /** General Context Mapper for DNs, which can be both, user or groups. */
+  class DnContextMapper extends AbstractContextMapper<AccessIdRepresentationModel> {
+
+    @Override
+    public AccessIdRepresentationModel doMapFromContext(final DirContextOperations context) {
+      final AccessIdRepresentationModel accessId = new AccessIdRepresentationModel();
+      String userId = context.getStringAttribute(getUserIdAttribute());
+      if (userId != null) {
+        accessId.setAccessId(userId);
+        String firstName = context.getStringAttribute(getUserFirstnameAttribute());
+        String lastName = context.getStringAttribute(getUserLastnameAttribute());
+        accessId.setName(String.format("%s, %s", lastName, firstName));
+      } else {
+        String dn = getDnWithBaseDn(context.getDn().toString());
+        accessId.setAccessId(dn); // fully qualified dn
+        accessId.setName(context.getStringAttribute(getGroupNameAttribute()));
+      }
```

Comment:
Suggestion: use the Method from UserContextMapper and GroupContextMapper by either extracting the content from their `doMapFromContext` into a dedicated method or by initalizing them here.

What do you think of this?